src/notification/email.ts

```diff
@@ -6,12 +6,23 @@ import nodemailer from 'nodemailer';
 
 const email = Config.notifications.email;
 
-const transporter = nodemailer.createTransport({
-	auth: {
-		pass: email.password,
-		user: email.username
-	},
-	service: 'gmail'
+const transportOptions: any = {};
+
+if (email.username && (email.password || email.smtpAddress)) {
+	transportOptions.auth = {};
+	transportOptions.auth.user = email.username;
+	transportOptions.auth.pass = email.password;
+}
```

Comment:
It's OK if password is empty?

```suggestion
if (email.username && (email.password || email.smtpAddress)) {
	transportOptions.auth = {};
	transportOptions.auth.username = email.username;
	transportOptions.auth.password = email.password;
}
```

Name change for consistency purposes.

Reply:
Yes, if an `smtpAddress` is configured, then a password is not required because some smtp relay agents don't require authentication (like the one I use). Also, `auth.user` and `auth.pass` are the attribute names that `createTransport` is expecting [nodemailer-Authentication](https://nodemailer.com/smtp/#authentication)  so changing them to `auth.username` and `auth.password` might break it.

Reply:
Makes sense to me then. No need to change.